src/middlewared/middlewared/plugins/vm.py

```diff
@@ -887,20 +887,51 @@ def random_mac(self):
         Int('vcpus'),
         Int('memory'),
         Str('bootloader'),
-        List("devices"),
+        List('devices'),
         Str('vm_type'),
         Bool('autostart'),
+        Bool('create_zvol', required=False),
+        Str('zvol_name', required=False),
+        Str('zvol_type', required=False),
+        Str('zvol_volsize', required=False),
```

Comment:
That seems counter-intuitive.

Why are the zvol options in the VM schema? I believe those would make more sense if they are part of the `devices` of type `DISK`?

That would also allow one to create multiple disks (zvols).

Reply:
Fair point :)